salt/master.py

```diff
@@ -2030,12 +2030,35 @@ def get_token(self, clear_load):
             return False
         return self.loadauth.get_tok(clear_load['token'])
 
+    def publish_batch(self, batch, clear_load):
+        batch_load = {}
+        batch_load.update(clear_load)
+        batch_load['kwargs']['batch'] = batch
+        import salt.cli.batch_async
+        batch = salt.cli.batch_async.BatchAsync(
+            self.local.opts,
+            functools.partial(self._prep_jid, clear_load, {}),
```

Comment:
Oh nice things you learn by doing PR reviews, `functools.partial`. Been looking for this for a while, thanks for sharing! :+1: